src/terraboot/core.clj

```diff
@@ -25,10 +25,28 @@
   (output-of type name "arn"))
 
 (defn data
-  [type name spec]
-  {:data
-   {type
-    {name spec}}})
+  ([type name spec]
+   {:data
+    {type
+     {name spec}}})
+  ([name-fn type name spec]
```

Comment:
I'd say it's more idiomatic to have the additional arity arguments at the end, rather than the beginning, of the arg list but there might be a valid reason for it.